rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -1477,7 +1477,7 @@ public ClassDeclaration withTypeParameters(@Nullable JContainer<TypeParameter> t
     @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
     @RequiredArgsConstructor
     @AllArgsConstructor(access = AccessLevel.PRIVATE)
-    final class CompilationUnit implements J, JavaSourceFile, SourceFile {
+    final class CompilationUnit implements J, JavaSourceFile, SourceFile, SourceFileWithReferences {
```

Comment:
Hmm; I've seen some conditionals checking first if something is a `JavaSourceFile`, and `else` if it is a `SourceFileWithReferences`. Those would have to be adjusted if we're to continue down this path.

Alternatively just have an option on ChangeType and ChangePackage to also include String literals (default false) could be the easier option.